Translation updates (not building)

twse opened the pull request:
This pull request does sentence casing on the Finnish and French translations and adds a Spanish translation.

humdingerb commented:
Looks good. What's still missing is the option to change to Spanish in `artpaint/windows/GlobalSetupWindow.cpp` and the flag in `artpaint/application/Flags.h`. Wanna work on that for this PR?

twse commented:
Hi humdinger. I added the options to change to Spanish and Finnish in `artpaint/windows/GlobalSetupWindow.cpp` and the flags in `artpaint/application/Flags.h`.

humdingerb commented on artpaint/windows/GlobalSetupWindow.cpp:
```diff
@@ -59,6 +59,7 @@ const uint32 kEnglishLanguageSet				= '_eng';
 const uint32 kFinishLanguageSet					= '_fin';
```
Should be "kFinnishLanguageSet"

humdingerb commented on artpaint/windows/GlobalSetupWindow.cpp:
```diff
@@ -404,6 +407,8 @@ GlobalSetupWindow::LanguageControlView::LanguageControlView()
 	fGerman = new BRadioButton("Deutsch", new BMessage(kGermanLanguageSet));
 	fFrench = new BRadioButton("Française", new BMessage(kFrenchLanguageSet));
+	fSpanish = new BRadioButton("Español", new BMessage(kSpanishLanguageSet));
+	fFinnish = new BRadioButton("Suomi", new BMessage(kFinishLanguageSet));
```
-> kFinnishLanguageSet

humdingerb commented on artpaint/windows/GlobalSetupWindow.cpp:
```diff
@@ -444,6 +449,12 @@ GlobalSetupWindow::LanguageControlView::LanguageControlView()
+		fSpanish->SetValue(B_CONTROL_ON);
+
+	if (fLanguage == FINNISH_LANGUAGE)
+		fFinish->SetValue(B_CONTROL_ON);
```
-> fFinnish

humdingerb commented on artpaint/windows/GlobalSetupWindow.cpp:
```diff
@@ -456,6 +467,8 @@ GlobalSetupWindow::LanguageControlView::AttachedToWindow()
 	fGerman->SetTarget(this);
 	fFrench->SetTarget(this);
+	fSpanish->SetTarget(this);
+	fFinish->SetTarget(this);
```
-> fFinnish

humdingerb commented on artpaint/windows/GlobalSetupWindow.cpp:
```diff
@@ -480,6 +493,13 @@ GlobalSetupWindow::LanguageControlView::MessageReceived(BMessage* message)
+			_Update(SPANISH_LANGUAGE);
+		}	break;
+
+		case kFinishLanguageSet: {
```
This case is already present further up... and there: -> kFinnishLanguageSet

twse commented:
Hi humdinger.

I uncommented the code for Finnish language setting, removed the duplicate code for kFinnishLanguageSet and corrected the language name (Finish to Finnish).

humdingerb commented:
We're missing return_finnish_string() in StringServer.cpp.
It'd help if you try to build the app before push your changes... :)

twse commented:
Hi all,
There's a return_finnish_string() in StringServer.cpp so I added it.

humdingerb commented:
It still doesn't build. Would you please test that it does before pushing?

All these constants seem to be missing from the FinnishStrings.h:
```

...StringServer.cpp: In function `static const char * StringServer::return_finnish_string(string_id)':
...StringServer.cpp:680: `FINNISH_ABOUT_ARTPAINT_STRING' undeclared (first use this function)
...StringServer.cpp:680: (Each undeclared identifier is reported only once
...StringServer.cpp:680: for each function it appears in.)
...StringServer.cpp:682: `FINNISH_ABOUT_HELP_STRING' undeclared (first use this function)
...StringServer.cpp:710: `FINNISH_ALPHA_STRING' undeclared (first use this function)
...StringServer.cpp:716: `FINNISH_BLUE_STRING' undeclared (first use this function)
...StringServer.cpp:720: `FINNISH_BRUSH_STRING' undeclared (first use this function)
...StringServer.cpp:722: `FINNISH_BRUSH_WINDOW_STRING' undeclared (first use this function)
...StringServer.cpp:726: `FINNISH_CANCEL_STRING' undeclared (first use this function)
...StringServer.cpp:734: `FINNISH_CHANGES_TAKE_EFFECT_STRING' undeclared (first use this function)
...StringServer.cpp:752: `FINNISH_CLICK_TO_SELECT_COLOR_STRING' undeclared (first use this function)
...StringServer.cpp:756: `FINNISH_COLOR_SET_STRING' undeclared (first use this function)
...StringServer.cpp:762: `FINNISH_COLOR_MODEL_STRING' undeclared (first use this function)
...StringServer.cpp:768: `FINNISH_COLOR_WINDOW_STRING' undeclared (first use this function)
...StringServer.cpp:772: `FINNISH_CONFIRM_QUIT_STRING' undeclared (first use this function)
...StringServer.cpp:786: `FINNISH_CROSS_HAIR_CURSOR_STRING' undeclared (first use this function)
...StringServer.cpp:788: `FINNISH_CURSOR_STRING' undeclared (first use this function)
...StringServer.cpp:800: `FINNISH_DO_NOT_SAVE_STRING' undeclared (first use this function)
...StringServer.cpp:806: `FINNISH_EFFECTS_WINDOW_STRING' undeclared (first use this function)
...StringServer.cpp:810: `FINNISH_EMPTY_PAINT_WINDOW_STRING' undeclared (first use this function)
...StringServer.cpp:852: `FINNISH_GREEN_STRING' undeclared (first use this function)
...StringServer.cpp:882: `FINNISH_KEEP_IN_FRONT_STRING' undeclared (first use this function)
...StringServer.cpp:892: `FINNISH_LAYER_WINDOW_STRING' undeclared (first use this function)
...StringServer.cpp:910: `FINNISH_MISCELLANEOUS_STRING' undeclared (first use this function)
...StringServer.cpp:932: `FINNISH_OPEN_COLOR_SET_STRING' undeclared (first use this function)
...StringServer.cpp:942: `FINNISH_OPEN_STRING' undeclared (first use this function)
...StringServer.cpp:960: `FINNISH_RECENT_IMAGES_STRING' undeclared (first use this function)
...StringServer.cpp:962: `FINNISH_RECENT_PROJECTS_STRING' undeclared (first use this function)
...StringServer.cpp:966: `FINNISH_RED_STRING' undeclared (first use this function)
...StringServer.cpp:974: `FINNISH_RELEASE_DATE_STRING' undeclared (first use this function)
...StringServer.cpp:988: `FINNISH_ROTATE_CW_STRING' undeclared (first use this function)
...StringServer.cpp:990: `FINNISH_ROTATE_CCW_STRING' undeclared (first use this function)
...StringServer.cpp:996: `FINNISH_SAVE_CHANGES_STRING' undeclared (first use this function)
...StringServer.cpp:998: `FINNISH_SAVE_COLOR_SET_AS_STRING' undeclared (first use this function)
...StringServer.cpp:1020: `FINNISH_SAVE_STRING' undeclared (first use this function)
...StringServer.cpp:1076: `FINNISH_STANDARD_SIZES_STRING' undeclared (first use this function)
...StringServer.cpp:1088: `FINNISH_TOOL_CURSOR_STRING' undeclared (first use this function)
...StringServer.cpp:1090: `FINNISH_TOOL_SELECTION_WINDOW_STRING' undeclared (first use this function)
...StringServer.cpp:1096: `FINNISH_TOOL_SETUP_WINDOW_STRING' undeclared (first use this function)
...StringServer.cpp:1120: `FINNISH_UNSUPPORTED_FILE_TYPE_STRING' undeclared (first use this function)
...StringServer.cpp:1256: `FINNISH_DO_CROP_HELP_STRING' undeclared (first use this function)
...StringServer.cpp:1258: `FINNISH_DO_ROTATE_HELP_STRING' undeclared (first use this function)
...StringServer.cpp:1260: `FINNISH_DO_SCALE_HELP_STRING' undeclared (first use this function)
...StringServer.cpp:1262: `FINNISH_DO_TRANSLATE_HELP_STRING' undeclared (first use this function)
...StringServer.cpp:1266: `FINNISH_ABOUT_1_TEXT_STRING' undeclared (first use this function)
...StringServer.cpp:1268: `FINNISH_ABOUT_3_TEXT_STRING' undeclared (first use this function)
...StringServer.cpp:1270: `FINNISH_ABOUT_4_TEXT_STRING' undeclared (first use this function)
...StringServer.cpp:1272: `FINNISH_ABOUT_5_TEXT_STRING' undeclared (first use this function)
...StringServer.cpp:1274: `FINNISH_ABOUT_6_TEXT_STRING' undeclared (first use this function)
...StringServer.cpp:1276: `FINNISH_ABOUT_7_TEXT_STRING' undeclared (first use this function)
...StringServer.cpp:1278: `FINNISH_ABOUT_8_TEXT_STRING' undeclared (first use this function)
...StringServer.cpp:1280: `FINNISH_ABOUT_9_TEXT_STRING' undeclared (first use this function)
...StringServer.cpp:1282: `FINNISH_ABOUT_10_TEXT_STRING' undeclared (first use this function)

```

twse commented:
Hi humdinger.

In order to completely build the program I translated the missing constants.

humdingerb commented:
Still doesn't build. Would you mind adding a "Doesn't build yet" to your commit messages? Then people don't spend time checking out the PR prematurely.

scottmc commented:
Can we get this one wrapped up?

humdingerb commented:
I wanted to add what was left and squash a few of its commits for this PR, but ended up with PR #77...
So, I merged #77 and am closing this #76. Thanks @twse who did all the work...
